### DISCUSSION TITLE
Is there anyway to get the OOB in json_key format?

### DISCUSSION THREAD

**Discussion by kaleemktk** [zsrv/supermicro-product-key#4] [General]
Is there anyway to get the OOB in json_key format?

**Comment by zsrv** [comment]
JSON keys cannot be generated, as Supermicro signs them using an RSA key.

**Reply by kaleemktk** [reply]
but we can do something like 
" go run main.go nonjson encode --sku SFT-DCMS-SINGLE 3cecef123456  "  
to get the DCMS License and then activate through SUM (SuperMicro Update Manager). Does this method work for X12+ motherboards? if yes, then is there anyway we can get the signature using that key. 
Thanks

**Reply by zsrv** [reply]
Unfortunately no. Supermicro signs the JSON keys with their RSA private key. The corresponding public key is embedded in the Supermicro firmware, and the signature is verified against that key. We can't generate JSON keys without Supermicro's private key.

**Reply by kaleemktk** [reply]
so does that mean that even using SUM and 
" go run main.go nonjson encode --sku SFT-DCMS-SINGLE 3cecef123456 " 
we cannot activate DCMS License?

**Reply by zsrv** [reply]
Correct. The non-JSON format is only accepted by 11th generation and older motherboards. There are a few 12th generation boards that use the non-JSON format; there is a link to the list of those boards in the README.

**Comment by stamepicmorg** [comment]
could we replace rsa pair from firmware?

**Reply by zsrv** [reply]
It may be possible (the public key is a string in libipmi.so) but one problem that comes to mind is that Supermicro started signing the firmware at some point. If a BMC has already been upgraded to a firmware that checks the signature on a firmware update that is being uploaded, this probably could not be done.
